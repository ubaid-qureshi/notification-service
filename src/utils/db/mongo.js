const mongoose = require('mongoose');
const config = require('config');
const { AppError } = require('../requestHandlers/errorHandler');

module.exports = async () => {
  try {
    mongoose.set('useFindAndModify', false);
    mongoose.pluralize(null);
    let url;
    if (process.env.NODE_ENV === 'test') {
      url = global.__MONGO_URI__;
    } else {
      mongoose.set('debug', process.env.NODE_ENV !== 'production');
      url = process.env.MONGODB_URI || config.db.MONGODB_URI;
    }
    await mongoose.connect(url, {
      useCreateIndex: true,
      useNewUrlParser: true,
      useUnifiedTopology: true,
    });
    mongoose.connection.on('error', (error) => {
      _logger.error(`Mongoose connection error: ${error}`);
      throw new AppError(400, error.message);
    });
    _logger.info(`mongodb database connection successful on ${url}`);
  } catch (error) {
    _logger.error(`database connection error- ${error.message}`);
    throw new AppError(400, error.message);
  }
};
