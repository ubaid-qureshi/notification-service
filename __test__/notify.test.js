require('../src/utils/logger/logConfig.js');
require('../src/utils/requestHandlers/responseHandler');
require('../src/utils/db/mongo')();
require('dotenv').config();
const mongoose = require('mongoose');

const user = require('../src/controller/User');
const notify = require('../src/controller/Notify');
const messaging = require('../src/controller/Messaging');

jest.mock('../src/controller/Messaging');

describe('Test suit for user', () => {
  beforeAll(async () => {
    await user.create({
      email: 'u.qureshi006@gmail.com',
      firstName: 'Ubaid',
      lastName: 'Qureshi',
      phoneNumber: '1234567890',
      city: 'mumbai',
      country: 'india',
    });
  });

  it('Notify should pass', async () => {
    messaging.sendToAllModes.mockResolvedValue({
      status: 'sent',
    });
    const response = await notify.notify(
      'all',
      'this is message',
      ['123456789'],
    );

    expect(response).toMatchSnapshot();
  });

  it('Notify should fail', async () => {
    try {
      messaging.sendToAllModes.mockResolvedValue({
        status: 'sent',
      });
      await notify.notify(
        undefined,
        'this is message',
        ['123456789'],
      );
    } catch (error) {
      expect(error.message).toBe('Please select correct notification mode');
    }
  });

  afterAll(async (done) => {
    await mongoose.connection.close();
    done();
  });
});
