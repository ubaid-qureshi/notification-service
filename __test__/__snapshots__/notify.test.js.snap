// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suit for user Notify should pass 1`] = `
Object {
  "status": "sent",
}
`;
